src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,130 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] storage;
+    private int currentCapacity;
+    private int size;
+
+    public MyHashMap() {
+        storage = new Node[INITIAL_CAPACITY];
+        currentCapacity = INITIAL_CAPACITY;
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (suchElementExists(key, value)) {
+            return;
+        }
+        int index = hashIndex(key);
+        if (storage[index] != null) {
+            Node<K, V> node = storage[index];
+            if (putWithKey(key, value, (Node<K, V>) node)) {
+                return;
+            }
+            if (!Objects.equals(node.key, key)) {
+                while (node.next != null) {
+                    node = node.next;
+                    if (putWithKey(key, value, node)) {
+                        return;
+                    }
+                }
+                linkLast(node, key, value);
+            }
+        } else {
+            storage[index] = new Node<>(key, value);
+        }
+        size++;
+        resize();
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> tempNode = storage[hashIndex(key)];
+        while (tempNode != null) {
+            if (Objects.equals(tempNode.key, key)) {
+                return tempNode.value;
+            }
+            tempNode = tempNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hashIndex(K key) {
+        return key != null ? Math.abs(key.hashCode() % currentCapacity) : 0;
+    }
+
+    private void resize() {
+        if (size > (storage.length * LOAD_FACTOR)) {
+            size = 0;
+            currentCapacity *= 2;
+            Node<K, V>[] tempStorage = storage;
+            storage = new Node[currentCapacity];
+            for (Node<K, V> tempNode : tempStorage) {
+                while (tempNode != null) {
+                    put(tempNode.key, tempNode.value);
+                    tempNode = tempNode.next;
+                }
+            }
+        }
+    }
+
+    private boolean putWithKey(K key, V value, Node<K, V> node) {
+        if (Objects.equals(node.key, key)) {
+            node.value = value;
+            return true;
+        }
+        return false;
+    }
+
+    private boolean suchElementExists(K key, V value) {
```

Comment:
this method makes code more complicated, try to remove it and complete whole logic in put method